status-im/status-desktop
# 15217 UI   wallet stability ensure  review all the entry points for send

### What does the PR do

closes #15217

This PR goes through all places where Send modal can be invoked from and fixes the identified issues.

General rules used when pre-filling the Send modal:
1. If the Send modal is launched in the context of an account, the account is pre-selected
2. If the Send modal is not launched in the context of an account, the first account in the list is pre-selected
3. If the Send modal is launched in the context of an asset, the asset is pre-selected
4. If the Send modal is launched in the context of a collectible, the collectible and owning account is pre-selected.
5. If the Send modal is launched in the context of a transaction, the account, receiver, token and about is pre-selected.


#### This PR does not cover:
"Repeat transaction" when an airdrop is selected. Will be handled in https://github.com/status-im/status-desktop/issues/12275

#### Changes:
1. [fix: Fixing crash on status-go exception when calling `getOwnedCollec…](https://github.com/status-im/status-desktop/commit/8839b4095de40104083dbf4a548034874d3e65ca) - Fixing a crash on status-go exception. It was happening when the account selection is empty
7. [chore(AccountSelector): Default to the first item in the account sele…](https://github.com/status-im/status-desktop/commit/0baafc1bef9b6d2517f4471e006be68ad630b9ac) - Update the account selector to default to the first item in the list if the pre-selected address is invalid (E.g. watch-only account)
8. https://github.com/status-im/status-desktop/commit/200917277164dbc4ecadbf85ab5ce067d1771c9e - Updating `SendModal` to accept account address for preselection instead of account model item. There are places in the app where the account model item is not available and it shouldn't be a prerequisite to forward the account model to every place where the send modal can be opened.
9. [fix(Send): Fine tune the Send action on collectibles](https://github.com/status-im/status-desktop/commit/2db2e2238cff60435aa144b8b0a4ea3ce93348f2) - Disable the `Send` button if the collectible is soul bound, hide the `Send` button if the collectible is owned by a watch-only account.
10. [fix(Send): Fixing Send button text on wallet footer on owner collecti…](https://github.com/status-im/status-desktop/commit/b622cd32f77e79d528b91a6d39e26241c5018cac) - The `Send` button text was not showing the community name. I've used the communities model to extract the name
11. [fix(SendModal): Fixing "Repeat transaction" action in Activity view a…](https://github.com/status-im/status-desktop/commit/82e55038ced995e3aa4831444addbf55db3a484c) - The repeat transaction was broken due to invalid amount parsing and wrong transaction processing before reaching the SendModal.

### Affected areas

SendModal - all the places where it can be launched from
Account selector
Buy sticker
Buy ENS

### Screenshot of functionality (including design for comparison)



#### Default account and token selection

https://github.com/user-attachments/assets/1259ef86-992e-4772-9f24-2872fba0c233

#### Prefilled receiver address

https://github.com/user-attachments/assets/a3afb187-2c85-4bc4-8950-e2cffecd43e0

#### Repeat transaction

https://github.com/user-attachments/assets/338b536a-bf7d-4f8f-a1e6-58ec8df1ff0c

#### Preselecting accounts and tokens when sending collectibles + Disable/hide Send button for soulbound/not owned tokens

https://github.com/user-attachments/assets/8b8ea3f4-6e75-49a6-9999-8f3ac9d87b83


<!-- Uncomment this section for status-go upgrade/dogfooding pull requests

### Impact on end user

What is the impact of these changes on the end user (before/after behaviour)

The user can launch Send modal from multiple places and the expect data is pre-filled based on context.

### How to test

- How should one proceed with testing this PR.
Go through all the places where Send can be invoked from and check if the preselected account, token, receiver address and amount match the preselection rules mentioned above:
1. Wallet footer (Assets, Collectibles, Asset details, Collectibles details, Activity, Activity details)
2. Assets/Collectibles context menu
3. Sent transaction context menu and Sent transaction details
4. Transfer ownership
- What kind of user flows should be checked?
1. Different wallet account layout configuration (E.g. The first account is watch-only, the first account is not watch-only, Changing the account order and re-launching the Send modal with different configurations)
2. Trying Send modal when `All accounts` is selected for assets, watch-only collectibles, watch-only community collectibles, owned collectibles, owned community collectibles, sent transactions
3. Transfer community ownership from `Community settings` and from `Wallet`

### Risk 

Described potential risks and worst case scenarios.

Tick **one**:
